server/src/main/java/org/candlepin/resource/RoleResource.java

```diff
@@ -283,14 +239,15 @@ public RoleDTO deleteUserFromRole(@PathParam("role_name") String roleName,
         return this.modelTranslator.translate(role, RoleDTO.class);
     }
 
-    @ApiOperation(notes = "Retrieves a list of Roles", value = "getRoles")
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    @Wrapped(element = "roles")
-    public Stream<RoleDTO> getRoles() {
+    /**
+     * {@InheritDoc}
+     */
+    @Override
+    public Iterable<RoleDTO> getRoles() {
         // TODO: Add in filter options
 
         Collection<? extends RoleInfo> roles = this.userService.listRoles();
-        return roles.stream().map(this.modelTranslator.getStreamMapper(RoleInfo.class, RoleDTO.class));
+        return roles.stream().map(this.modelTranslator.getStreamMapper(RoleInfo.class, RoleDTO.class))
+            .collect(Collectors.toSet());
```

Comment:
There's no reason to change this into returning a Set. We can still return a Stream here, by the x-java-response: type: to `java.util.stream.Stream` from `Iterable` (check other examples in the spec file)

Reply:
Fixed it.